src/prefs/DirectoriesPrefs.cpp

```diff
@@ -377,7 +378,8 @@ void DirectoriesPrefs::OnBrowse(wxCommandEvent &evt)
       }
    }
 
-   if (!FileNames::WritableLocationCheck(dlog.GetPath()))
+   //Checks if the location for Open,Save.Import,Export and Macro Output has write permissions(Browse Buttons)
+   if (!FileNames::WritableLocationCheck(dlog.GetPath(), wxT("Cannot set the preference,")))
```

Comment:
This common line is for Open, Save, Import, Export, and Macro output directories.  Each should have a different informative string.  Make a table of strings and index into it with `id`.  Make each string a full sentence, even though that is repetitious and only changes one word each time.  Don't make one string that you substitute a word into -- that may not translate well into all languages.